lib/engine/game/g_1880/step/buy_train.rb

```diff
@@ -55,7 +58,14 @@ def discard_trains?
           def discard_all_trains(train_name)
             @game.log << "#{train_name} has not been bought for a full operating order, "\
                          "removing all remaining #{train_name} trains"
-            @game.depot.export_all!(train_name, silent: true)
+            return @game.depot.export_all!(train_name, silent: true) unless special_edge_case_4t_rocket?(train_name)
+
+            @game.depot.export!(train_name, silent: true)
+            @round.special_edge_case_4t_rocket = true
```

Comment:
I think this is a bit more readable, and then I don't think anything here is that ugly:

```suggestion
            if special_edge_case_4t_rocket?(train_name)
              @game.depot.export!(train_name, silent: true)
              @round.special_edge_case_4t_rocket = true
            else
              @game.depot.export_all!(train_name, silent: true) unless 
            end
```

Reply:
The ugliness is that I'm exporting one train, and relying on another event defined somewhere else in the code to kick in, take over and handle the rest. 
It makes for hard to maintain/debug code. It's not obvious what's happening here, unless you know the inner working of the system and specifically the 1880 assign.rb class.

Reply:
Alright, I'm not very familiar with 1880 so I'll leave it to @crericha for code review suggestions.